src/Transport/Topology/MetaModel/NamespaceBundleConfigurations.cs

```diff
@@ -0,0 +1,44 @@
+﻿namespace NServiceBus.AzureServiceBus.Topology.MetaModel
+{
+    using System;
+    using System.Collections;
+    using System.Collections.Generic;
+    using System.Linq;
+
+    class NamespaceBundleConfigurations : IEnumerable<NamespaceBundleInfo>
+    {
+        List<NamespaceBundleInfo> namespaceBundles;
+
+        public NamespaceBundleConfigurations()
+        {
+            namespaceBundles = new List<NamespaceBundleInfo>();
+        }
+
+        public void Add(string namespaceAlias, int numberOfTopicsInBundle)
+        {
+            var bundleInfo = namespaceBundles.SingleOrDefault(x => x.Alias.Equals(namespaceAlias, StringComparison.OrdinalIgnoreCase));
+            if (bundleInfo != null && bundleInfo.NumberOfTopicsInBundle < numberOfTopicsInBundle)
+            {
+                throw new Exception($"Duplicate namespace alias `{namespaceAlias}` was added to the {nameof(NamespaceBundleConfigurations)}. Namespace aliases should be unique.");
+            }
+
+            namespaceBundles.Add(new NamespaceBundleInfo(namespaceAlias, numberOfTopicsInBundle));
+        }
+
+        public int GetNumberOfTopicInBundle(string namespaceAlias)
+        {
+            var selected = namespaceBundles.SingleOrDefault(x => x.Alias.Equals(namespaceAlias, StringComparison.OrdinalIgnoreCase));
+            return selected?.NumberOfTopicsInBundle ?? 0;
```

Comment:
I disagree, this method should return the best value possible and not assume that it's caller will fix it. 0 is a bad value